### PR TITLE
Take wrapping into account when expanding wordwise selections

### LINKED ISSUE #17165: Double-click word selection picks up the RPROMPT from the previous line because word wrapping is hard
Discussed this with Leonard a while ago

### PR DESCRIPTION
**Pull request by DHowett**
Closes #17165

### CODE REVIEW

File: src/buffer/out/textBuffer.cpp [hunk 1]
```diff
@@ -1440,10 +1440,23 @@ til::point TextBuffer::_GetWordStartForSelection(const til::point target, const
...
-        //prevent selection wrapping on whitespace selection
-        if (isControlChar && result.x == bufferSize.Left())
+        if (result.x == bufferSize.Left())
         {
-            break;
+            // Prevent wrapping to the previous line if the selection begins on whitespace
+            if (isControlChar)
+            {
+                break;
+            }
```
Comment (lhecker) [inline src/buffer/out/textBuffer.cpp:1449]: ...but should we? If I print "a    b" and it force-wraps in the middle, shouldn't I be able to select all 4 spaces with a double click?

**Review by lhecker — APPROVED** [review]

File: src/buffer/out/textBuffer.cpp (hunk 1 above)
Comment (DHowett) [reply src/buffer/out/textBuffer.cpp:1449]: Eh, I didn't want to revisit too many of the decisions from the previous code.

But also, this is what prevents us from selecting **the entire text buffer** when you double-click an empty space lmao

File: src/host/ut_host/TextBufferTests.cpp [hunk 2]
```diff
@@ -2336,16 +2336,32 @@ void TextBufferTests::GetWordBoundaries()
...
+    // 1|swrapped  | < not wrapped
+    // 2|notwrapped| < not wrapped
+    // 3|spaces    | < wrapped
+    // 4|    wrappe| < wrapped
+    // 5|d reachEOB| < wrapped
+
+    VERIFY_IS_TRUE(_buffer->GetRowByOffset(0).WasWrapForced());
+    VERIFY_IS_FALSE(_buffer->GetRowByOffset(1).WasWrapForced());
+    // GH#780 See the comment in WriteLinesToBuffer
+    // VERIFY_IS_FALSE(_buffer->GetRowByOffset(2).WasWrapForced());
```
Comment (carlos-zamora) [inline src/host/ut_host/TextBufferTests.cpp:2358]:
```suggestion
```
dead code?

**Review by carlos-zamora — APPROVED** [review]

File: src/host/ut_host/TextBufferTests.cpp (hunk 2 above)
Comment (DHowett) [reply src/host/ut_host/TextBufferTests.cpp:2358]: No, pertinent to the comment above it

**Review by zadjii-msft — APPROVED** [review]

**Review by Zionsammy — COMMENTED** [review]
Good

**Review by Zionsammy — COMMENTED** [review]
Thanks

**Pull request merged**